### DISCUSSION TITLE
My Fooocus takes 10-15 minutes to generate images please help.

### DISCUSSION THREAD

**Discussion by yusiff** [lllyasviel/Fooocus#3523] [Q&A]
It takes Fooocus 10-15 minutes to generate images regardless of how long/short the prompt is and I have no clue what the issue is

**Device specs:**
Processor	Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz   2.50 GHz
Installed RAM	32.0 GB (31.8 GB usable)
System type	64-bit operating system, x64-based processor
Windows 10

**This is what I get upon launching run.bat**
[Cleanup] Cleanup successful
Total VRAM 4096 MB, total RAM 32569 MB
Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --always-normal-vram
Set vram state to: LOW_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce GTX 1650 Ti : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
Running on local URL:

Appreciate the help!

**Comment by DavidDragonsage** [comment]
Hey - you're using my computer! Our video cards do not meet the minimum requirements for running Fooocus, as listed here:
https://github.com/lllyasviel/Fooocus
(scroll about 3/4 way down the page)

Yes, image generation is damn slow! Although once you start a generation series, the succeeding images are considerably faster. Nonetheless I enjoy using Fooocus, and until the release of Forge it was the only SDXL UI capable of running on my PC.

It will not matter at all how long the prompt is. Try to keep other windows and apps to a minimum while running Fooocus because they can be eating up your precious and small VRAM.

The type and speed of "x64-based processor" matters too, My CPU is rated about 14% faster but the big bottle-neck for both of us is the video card. I am saving to buy a new video card.

But also, Fooocus should be running on an NVME Solid State Drive. Running it from a hard drive will make it extra slow. A new NVME drive costs in the order of about C$150 (Canadian dollars). I have a small NVME drive that does not have the room to hold all the base models that I use (it only holds two of them). But it does hold Fooocus itself, and that is vital.

**Reply by yusiff** [reply]
Hey! Thanks for your reply. Don't you think the issue is with GPU as mentioned in the crm?
(Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --always-normal-vram)
Do you know how I can increase that?

Also, I checked the link (https://github.com/lllyasviel/Fooocus) and it says that Nvidia GTX 1XXX 8GB (* 6GB uncertain) is compatible. Since my video card is NVIDIA GeForce GTX 1650 Ti - doesn't it make it compatible?

Regards

**Reply by Rogala** [reply]
Your video card is compatible and it will work, but its parameters are very low, so the time to create one image is very long.

I have a 1650 super, on Speed mode the image is created for about 6 minutes.

**Comment by DavidDragonsage** [comment]
@yusiff Yes, the video card (GPU) is the primary problem. While the NVidia 1XXX is compatible in that it uses the NVidia technology that Fooocus is optimized for, with only 4GB of VRAM is does not meet the minimum requirements as stated in the link. Note that even 6GB is "uncertain" and at 8GB VRAM, the GTX 1XXX is "only marginally faster than CPU".

I know of no practical way to expand the VRAM on these cards. As an electronics technologist it might be possible and technically feasible to unsolder the VRAM chips and solder in larger ones, but I would not willingly attempt this on what is really a disposable consumer item. 

Replacement of the video card is the best solution, provided that your computer's power supply can handle the extra load. Unfortunately that is another possible problem, and power supplies are not always readily upgradeable in some PCs (such as mine), If and when I upgrade my video card, I know that could be a risk. My wife has a 12GB NVidia card in her PC, which runs Fooocus very nicely. The plan is to upgrade hers to a 16GB platform, then I will try her old card in my PC and see what happens.

Even with our low VRAM cards, 15 mins. per generation does seem excessive. Is it a lot faster once you have done the first generation in a series? If I ever experience an unusually long generation time I reboot the PC and that resolves the problem. But in general I do not measure generation times. I just do something else until the images are ready.

But about that other issue that I mentioned, did you check if Fooocus is running from an NVME Solid State Drive? If it is running from a hard drive that will definitely contribute to long generation times.

**Comment by DavidDragonsage** [comment]
@yusiff  I just noticed you might be asking about that --always-normal-vram switch that can be placed in your startup batch file, run.bat.

Do not do this, it will not help. Allow Fooocus to detect that you have only 4GB of VRAM. This optimizes Fooocus for a low VRAM environment.
